### PR TITLE
[11.x] Simplify `.php` extension chopping in `getNameInput`

### PR DESCRIPTION
**Pull request by osbre**

### CODE REVIEW

File: src/Illuminate/Console/GeneratorCommand.php [hunk 1]
```diff
@@ -417,13 +417,7 @@ protected function sortImports($stub)
...
     protected function getNameInput()
     {
-        $name = trim($this->argument('name'));
-
-        if (Str::endsWith($name, '.php')) {
-            return Str::substr($name, 0, -4);
-        }
-
-        return $name;
+        return (string) Str::of($this->argument('name'))->trim()->beforeLast('.php');
```
Comment (gianfranco-rocco) [inline src/Illuminate/Console/GeneratorCommand.php:420]: You could've called `toString()` and that would've passed the result to a string value.

Comment (osbre) [reply src/Illuminate/Console/GeneratorCommand.php:420]: @gianfranco-rocco it's `__toString()`  not `toString()`, so would've looked less clean.

Comment (gianfranco-rocco) [reply src/Illuminate/Console/GeneratorCommand.php:420]: There's actually a `toString` method on the `Stringable` class. Here's the link to the [API doc](https://laravel.com/api/11.x/Illuminate/Support/Stringable.html#method_toString).

**Comment by jasonmccreary** [comment]
While very edgy, it's worth noting these aren't logically equivalent. Consider passing a `make` command an odd path like: `tests/Feature/fixtures.php/SomeTest`.

With this change, `name` is, incorrectly, `tests/Feature/fixtures`. Before it was, correctly, `tests/Feature/fixtures.php/SomeTest`.

**Comment by osbre** [comment]
Thanks @jasonmccreary, sent a fix: https://github.com/laravel/framework/pull/51909

**Comment by driesvints** [comment]
This one was reverted.

**Pull request merged**